app/org/sagebionetworks/bridge/services/SurveyService.java

```diff
@@ -99,6 +104,16 @@ public Survey createSurvey(Survey survey) {
             dataGroups = study.getDataGroups();
         }
         Validate.entityThrowingException(new SurveySaveValidator(dataGroups), survey);
+
+        // Check survey ID uniqueness.
+        StudyIdentifier studyId = new StudyIdentifierImpl(survey.getStudyIdentifier());
+        String existingSurveyGuid = surveyDao.getSurveyGuidForIdentifier(studyId, survey.getIdentifier());
+        if (existingSurveyGuid != null) {
+            String errMsg = "Survey identifier " + survey.getIdentifier() + " is already used by survey " +
+                    existingSurveyGuid;
+            Map<String, Object> entityKeyMap = ImmutableMap.of(KEY_IDENTIFIER, survey.getIdentifier());
+            throw new EntityAlreadyExistsException(Survey.class, errMsg, entityKeyMap);
+        }
```

Comment:
To my mind, this kind of a constraint would be thrown before we'd throw a validation exception.

Reply:
Moved this to before the validation check.